src/test/rules/PreferMcAgarTest.ts

```diff
@@ -0,0 +1,151 @@
+    {
+      filename: 'src/test/ts/MyTest.ts',
+      code: `
+      import { TinyDom } from '@ephox/mcagar';
+      const doc = TinyDom.document(editor);
+      const body = TinyDom.body(editor);
+      const container = TinyDom.container(editor);
+      const contentArea = TinyDom.contentAreaContainer(editor);
+      const target = TinyDom.targetElement(editor);
+      `
+    }
+  ],
+  invalid: [
+    {
+      filename: 'src/test/ts/MyTest.ts',
+      code: `
+      import assert from 'chai';
+      assert.equal(ed.getContent(), '<p></p>');
+      assert.equal(editor.getContent(), '<p></p>');
+      assert.deepEqual(editor.getContent(), '<p></p>');
+      assert.equal(editor.getContent({ format: 'html' }), '<p></p>');
+      assert.equal(editor.getContent({ no_events: true }), '<p></p>');
+      `,
+      errors: [
+        { message: 'Use `TinyAssertions.assertContent` instead of manually asserting the content.' },
+        { message: 'Use `TinyAssertions.assertContent` instead of manually asserting the content.' },
+        { message: 'Use `TinyAssertions.assertContent` instead of manually asserting the content.' },
+        { message: 'Use `TinyAssertions.assertContent` instead of manually asserting the content.' },
+        { message: 'Use `TinyAssertions.assertContent` instead of manually asserting the content.' }
+      ],
+      // Note: The fixer doesn't touch imports
+      output: `
+      import assert from 'chai';
+      TinyAssertions.assertContent(ed, '<p></p>');
+      TinyAssertions.assertContent(editor, '<p></p>');
+      TinyAssertions.assertContent(editor, '<p></p>');
+      TinyAssertions.assertContent(editor, '<p></p>', { format: 'html' });
+      TinyAssertions.assertContent(editor, '<p></p>', { no_events: true });
+      `
+    },
+    {
+      filename: 'src/test/ts/MyTest.ts',
+      code: `
+      import assert from 'chai';
+      assert.equal(
+        ed.getContent(),
+        '<p></p>'
+      );
+      assert.equal(
+        ed.getContent({ no_events: true }),
+        '<p></p>'
+      );
+      `,
+      errors: [
+        { message: 'Use `TinyAssertions.assertContent` instead of manually asserting the content.' },
+        { message: 'Use `TinyAssertions.assertContent` instead of manually asserting the content.' }
+      ],
+      // Note: The fixer doesn't touch imports
+      output: `
+      import assert from 'chai';
+      TinyAssertions.assertContent(
+        ed,
+        '<p></p>'
+      );
+      TinyAssertions.assertContent(
+        ed,
+        '<p></p>',
+        { no_events: true }
+      );
+      `
+    },
+    {
+      filename: 'src/test/ts/MyTest.ts',
+      code: `
+      import assert from 'chai';
+      assert.equal(ed.getContent({ format: 'raw' }), '<p></p>');
+      assert.equal(editor.getContent({ format: 'raw' }), '<p></p>');
+      assert.deepEqual(editor.getContent({ format: 'raw' }), '<p></p>');
+      assert.equal(editor.getContent({ format: 'raw', no_events: true }), '<p></p>');
```

Comment:
This one needs tweaks to `assertRawContent` also `assertRawContent` is doing body.innerHTML not editor.getContent({format: 'raw') we should really change that since it's not the same thing.

- [ ] Consider skipping raw assertions check now since it needs changes to mcagar and possibly some tests

Reply:
We still want people to be using `assertRawContent` for any assertions, so even though it's different and might require some tweaking of the assertion we still want people to use that. I'll change it though so it only matches when there's only the single `format: 'raw'` property is specified though.

Reply:
I should add if we want a proper raw assertion than `body.innerHTML` is more accurate, so IMO that is correct anyways.

Reply:
Okay, I've done as I said above and also added checks to catch the `editor.getBody().innerHTML` version of the raw content assertion.

Reply:
Merging now, but if you still want something changed let me know Spocke.